msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/SynopticMsgcntrManagerImpl.java

```diff
@@ -70,48 +80,68 @@ public class SynopticMsgcntrManagerImpl extends HibernateDaoSupport implements S
 	private HashMap mfPageInSiteMap, sitesMap;
 	// transient variable for when on home page of site
 	private transient DecoratedCompiledMessageStats siteContents;
+
 	private MessageForumsMessageManager messageManager;
 	private UIPermissionsManager uiPermissionsManager;
+	private PreferencesService preferencesService;
+
+	@Getter
 	private PrivateMessageManager pvtMessageManager;
+
 	private MessageForumsTypeManager typeManager;
 	private DiscussionForumManager forumManager;
-	
+
 	/* Kernel */
 	private SqlService sqlService;
 	private SecurityService securityService;
 	private SiteService siteService;
 
-	public void setSecurityService(SecurityService securityService) {
-		this.securityService = securityService;
-	}
-
-	public void setSiteService(SiteService siteService) {
-		this.siteService = siteService;
-	}
-
-	public void setSqlService(SqlService sqlService) {
-		this.sqlService = sqlService;
-	}
-	
 	private static int ORACLE_IN_CLAUSE_SIZE_LIMIT = 1000;
 
-	
 	public SynopticMsgcntrManagerImpl() {}
-	
+
 	public void init() {
 		log.info("init()");
 	}
 
-	public List<SynopticMsgcntrItem> getWorkspaceSynopticMsgcntrItems(final String userId) {
+    public List<SynopticMsgcntrItem> getWorkspaceSynopticMsgcntrItems(final String userId) {
 
-		HibernateCallback<List<SynopticMsgcntrItem>> hcb = session -> {
+        HibernateCallback<List<SynopticMsgcntrItem>> hcb = session -> {
             Query q = session.getNamedQuery(QUERY_WORKSPACE_SYNOPTIC_ITEMS);
             q.setParameter("userId", userId, StringType.INSTANCE);
             return q.list();
         };
 
-		return getHibernateTemplate().execute(hcb);
-	}
+        List<SynopticMsgcntrItem> items = getHibernateTemplate().execute(hcb);
+
+        Preferences prefs = preferencesService.getPreferences(userId);
+        if (prefs != null) {
+            ResourceProperties props = prefs.getProperties(PreferencesService.SITENAV_PREFS_KEY);
+            if (props != null) {
+                int tabLabelPref = NumberUtils.toInt(props.getProperty(PreferencesService.TAB_LABEL_PREF), -1);
+                if (tabLabelPref == 2) {
+                    // This user wants the site description, not the title
+                    items.forEach(item -> {
+
+                        try {
+                            Site site = siteService.getSite(item.getSiteId());
+                            if (StringUtils.isNotBlank(site.getShortDescription())) {
+                                item.setSiteDescription(site.getShortDescription());
+                            }
+                        } catch (IdUnusedException e) {
+                            log.warn("No site for id {}", item.getSiteId());
+                        }
+                    });
+
+                    return items;
+                }
+            }
+        } else {
+            return items;
+        }
+
+        return getHibernateTemplate().execute(hcb);
```

Comment:
Do we want to re-run the query when `props == null || tabLabelPref != 2` or should we remove the else and `return items`?